2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,476 @@
+
+A `-Filter` parameter will accept a string array and will filter based on all PSResource metadata (including resource descriptions).
+
+A `-Proxy` parameter will accept a uri.
+
+A `-ProxyCredential` parameter will accept a PSCredential
+
+A `-Credential` parameter will accept a PSCredential.
+
+An `-IncludeDependencies` switch will include dependent resources in the returned results.
+
+A `-NoClobber` switch will prevent installing modules that have the same cmdlets
+as a differently named module already on the system.
+
+A `-Quiet` switch will suppress progress information.
+
+`-Scope` with `AllUsers` and `CurrentUser` will work the same as it does in v2.
+Default is `CurrentUser`.
+
+If aliases are enabled `Install-Module` will call to `Install-PSResource -Type Module`.
+
+v3 will continue to use the versioning folder format as v2 in that it will be
+Major.Minor.Patch and not contain the semver prerelease label.
+
+### Dependencies and version management
+
+`Install-PSResource` can accept a path to a psd1 or json file (using `-RequiredResourceFile`),
+or a hashtable or json (using `-RequiredResource`) where the key is the module name and the
+value is either the required version specified using Nuget version range syntax or
+a hash table where `repository` is set to the URL of the repository and
+`version` contains the [Nuget version range syntax](https://docs.microsoft.com/en-us/nuget/reference/package-versioning#version-ranges-and-wildcards).
+
+```powershell
+Install-PSResource -RequiredResource @{
+  'Configuration' = '[1.3.1,2.0)'
+  'Pester'        = @{
+    version = '[4.4.2,4.7.0]'
+    repository = 'https://www.powershellgallery.com'
+    credential = $cred
+    allowPrerelease = $true
+  }
+}
+```
+In this case the modules named "Configuration", and "Pester" will be installed. 
+The json format will be the same as if this hashtable is passed to `ConvertTo-Json`:
+
+```json
+{
+  'Configuration': '[1.3.1,2.0)',
+  'Pester': {
+    'version': '[4.4.2,4.7.0]',
+    'credential': null,
+    'repository': 'https://www.powershellgallery.com',
+    'prerelease': true
+  }
+}
+```
+
+>[!NOTE]
+>Credentials management will be using [Secrets Management RFC](https://github.com/PowerShell/PowerShell-RFC/pull/208)
+> for generalized credential management.
+The `repository` property can be the name of the registered repository or the URL
+to a repository.
+If no `repository` is specified, it will use the `default` repository.
+
+The older `System.Version` four part version type will be supported to retain
+compatibility with existing published modules using that format.
+
+If the resource requires dependencies that are not already installed, then
+a prompt will appear _before anything is installed_, listing all the resources
+including module name, version, size, and repository to be installed unless
+`-IncludeDependencies` is specified which will install without prompting.
+Rejecting the prompt will result in nothing being installed.

```

Comment:
This doc talks about resources that have dependencies but does not mention how will we handle modules that depend on a nuget library.

If we can define the nuget dependency in the published module, what happens when the user installs the module with out a `-DestinationPath`? Are they prevented from installing it because they can't save the nuget package? How would the user know how to handle that situation or should they be responsible to solve it? Or if they do specify a `-DestinationPath` that is their module folder and end up with libraries all over the root of the folder? 

If they do install with the module, where do they go? What would module authors have to do to use those libraries in that module? Will they drop into a standard shared location that's in the `$env:path`  or a module's lib folder that can be referenced with the psd1? Or would they have to hand craft the loading logic in the psm1 or the scripts to process using?

Or is this out of scope and modules authors will continue to embed the libraries inside their module like they do today and distribute it as one (large) module? Or download them on demand from the psm1 with `Install-PSResource`?

Reply:
Some built in handling of nuget dependencies would be a huge benefit, would _love_ to see that.